simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/support/CertificateUtils.java

```diff
@@ -0,0 +1,136 @@
+import java.security.*;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class CertificateUtils {
+
+    private static final String SIGNATURE_ALG = "SHA256WITHECDSA";
+
+    public static List<X509Certificate> convertStringToCertificates(String chain) {
+        try {
+            byte[] bytes = chain.getBytes();
+            CertificateFactory factory = CertificateFactory.getInstance("X.509");
+            InputStream in = new ByteArrayInputStream(bytes);
+            return factory.generateCertificates(in).stream().map(certificate -> ((X509Certificate) certificate)).collect(Collectors.toList());
+        } catch (Exception e) {
+            log.debug("Invalid certificate", e);
+        }
+        return Collections.emptyList();
+    }
+
+    public static boolean isCertificateValid(X509Certificate certificate, boolean isHardwareCertificate) {
+        try {
+            if(!isHardwareCertificate){
+                certificate.checkValidity();
+            }
+            String serialCode = StringUtils.removeStart(certificate.getSubjectDN().getName(), "CN=");
+            if (!StationHardwareData.SERIAL_NUMBER.equals(serialCode)) {
+                return false;
+            }
+        } catch (Exception e) {
+            log.debug("Exception while checking certificate validity", e);
+            return false;
+        }
+        return true;
+    }
+
+    public static String loadCertificateChain(String path){
+        StringBuilder certificateChain = new StringBuilder();
+        try (FileInputStream fis = new FileInputStream(path);
+             InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8);
+             BufferedReader reader = new BufferedReader(isr)
+        ) {
+            certificateChain.append(reader.lines().collect(Collectors.joining(System.lineSeparator())));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return certificateChain.toString();
+    }
+
+    public static String generatePKCS10(PublicKey publicKey, PrivateKey privateKey) throws IOException, OperatorCreationException {
+        X500Principal principal = new X500Principal( "CN=" + StationHardwareData.SERIAL_NUMBER);
+        PKCS10CertificationRequestBuilder p10Builder = new JcaPKCS10CertificationRequestBuilder(principal, publicKey);
+        JcaContentSignerBuilder csBuilder = new JcaContentSignerBuilder(SIGNATURE_ALG);
+        ContentSigner signer = csBuilder.build(privateKey);
+        PKCS10CertificationRequest csr = p10Builder.build(signer);
+        try (StringWriter stringWriter = new StringWriter()){
+            try (JcaPEMWriter pemWriter = new JcaPEMWriter(stringWriter)){
```

Comment:
One try block should be enough
try (StringWriter stringWriter = new StringWriter(); JcaPEMWriter pemWriter = new JcaPEMWriter(stringWriter))